### PR TITLE
Angular. Add Validators on Form

### PR DESCRIPTION
**Pull request by GoodDayForSurf**

### CODE REVIEW

File: packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.html [hunk 1]
```diff
@@ -191,25 +199,27 @@
...
-      </div>
+        <div *dxTemplate="let data of 'email'">
+          <form-item-plain
+            [isEditing]="isEditing"
+            [label]="data.editorOptions.label"
+            [value]="contactData$.email"
+            [validators]="[{type: 'email', message: 'Email is invalid'}] "
+          ></form-item-plain>
+          <dx-button class="form-item-plain-button"
+                     [visible]="!isEditing"
```
Comment (dxvladislavvolkov) [inline packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.html:210]: Fix indentation

File: packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
 import {
-  Component, OnInit, NgModule, Input, OnDestroy,
+  Component, OnInit, NgModule, Input, OnDestroy, ViewChild,
 } from '@angular/core';
 import { CommonModule } from '@angular/common';
+import ValidationEngine from 'devextreme/ui/validation_engine';
 import {
-  DxButtonModule,
+  DxButtonModule, DxFormComponent,
```
Comment (dxvladislavvolkov) [inline packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.ts:7]: Move to new line

File: packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.ts [hunk 3]
```diff
@@ -55,7 +66,13 @@ export class ContactFormComponent implements OnInit, OnDestroy {
     this.contactSubscription.unsubscribe();
   }
 
+  checkValidation = (e?) => {
```
Comment (dxvladislavvolkov) [inline packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.ts:69]: Remove the argument

File: packages/angular/src/app/shared/components/form-item-plain/form-item-plain.component.ts [hunk 4]
```diff
@@ -30,7 +30,15 @@ import { CommonModule } from '@angular/common';
         [value]="value"
         (onValueChanged)="onChangeValue($event)"
         [mask]="mask"
-      ></dx-text-box>
+      >
+        <dx-validator>
+          <dxi-validation-rule *ngFor="let rule of (validators || [])"
+                               [type]="rule.type"
```
Comment (dxvladislavvolkov) [inline packages/angular/src/app/shared/components/form-item-plain/form-item-plain.component.ts:36]: Fix indentation

**Review by dxvladislavvolkov — CHANGES_REQUESTED** [review]

File: packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.ts [hunk 5]
```diff
@@ -38,10 +48,14 @@ export class ContactFormComponent implements OnInit, OnDestroy {
...
   stylingMode = 'underlined';
 
-  editorOptions = { stylingMode: this.stylingMode };
+  isFormValid = true;
+
+  editorOptions: Record<string, any> = { stylingMode: this.stylingMode };
 
   contactSubscription: Subscription = new Subscription();
 
+  zipCodeValidator = { type: 'pattern', pattern: /\d{5}/, message: '5 digits' };
```
Comment (MikeVitik) [inline packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.ts:57]: Please check the message to our help writers

**Review by MikeVitik — CHANGES_REQUESTED** [review]

File: packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.html [hunk 6]
```diff
@@ -7,209 +7,180 @@
...
-                   [type]="'default'"
-                   stylingMode="outlined"
-        ></dx-button>
-      </div>
+        <dxi-item [colSpan]="2">
+          <edit-view-item label="Email"
+                           [isEditing]="isEditing"
+                           [value]="contactData$.email"
+                           [validators]="[{type: 'email', message: 'Email is invalid'}] "
+          ></edit-view-item>
```
Comment (MikeVitik) [inline packages/angular/src/app/pages/crm-contact-details/contact-form/contact-form.component.html:167]: Remove message

**Review by Shuvaevmedia — COMMENTED** [review]
Try placing the icon inside the field.

<dx-text-box
        value="password"
        placeholder="password"
        stylingMode="filled"
        [(mode)]="passwordMode"
      >
        <dxi-button
          name="today"
          location="before"
          [options]="passwordButton"
        ></dxi-button>
        
        https://js.devexpress.com/Demos/WidgetsGallery/Demo/Common/CustomTextEditorButtons/Angular/Light/

**Review by Shuvaevmedia — APPROVED** [review]

**Review by MikeVitik — APPROVED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Pull request merged**